src/execution/mod.rs

```diff
@@ -104,48 +104,58 @@ impl CallInfo {
         )
     }
 
-    /// Yields the contract calls in DFS (preorder).
+    /// Returns the contract calls in DFS (preorder).
     pub fn gen_call_topology(&self) -> Vec<CallInfo> {
         let mut calls = Vec::new();
+        calls.push(self.clone());
         if self.internal_calls.is_empty() {
-            calls.push(self.clone())
+            return calls;
         } else {
-            calls.push(self.clone());
             for call_info in self.internal_calls.clone() {
                 calls.extend(call_info.gen_call_topology());
             }
         }
+
         calls
     }
 
-    /// Returns a list of Starknet Event objects collected during the execution, sorted by the order
+    /// Returns a list of [`Event`] objects collected during the execution, sorted by the order
     /// in which they were emitted.
     pub fn get_sorted_events(&self) -> Result<Vec<Event>, TransactionError> {
+        // collect a vector of the full call topology (all the internal 
+        // calls performed during the current call)
         let calls = self.gen_call_topology();
-        let n_events = calls.iter().fold(0, |acc, c| acc + c.events.len());
-
-        let mut starknet_events: Vec<Option<Event>> = (0..n_events).map(|_| None).collect();
-
-        for call in calls {
-            for ordered_event in call.events {
-                let event = Event::new(ordered_event.clone(), call.contract_address.clone());
-                starknet_events.remove((ordered_event.order as isize - 1).max(0) as usize);
-                starknet_events.insert(
-                    (ordered_event.order as isize - 1).max(0) as usize,
-                    Some(event),
-                );
+        let mut collected_events = Vec::new();
+        
+        // for each call, collect its ordered events
+        for c in calls {
+            collected_events.extend(c.events.iter().map(|oe| (oe.clone(), c.contract_address.clone())));
+        }
+        // sort the collected events using the ordering given by the order
+        collected_events.sort_by_key(|(oe, _)| oe.order);
+        
+        // check that there is no holes.
+        // Since it is already sorted, we only need to check for continuity
```

Comment:
Nit: for consistency, start all either with lowercase or uppercase:
```suggestion
        // since it is already sorted, we only need to check for continuity
```